bitnami/prometheus/templates/server/deployment.yaml

```diff
@@ -0,0 +1,277 @@
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.server.customLivenessProbe "context" $) | nindent 12 }}
+          {{- else if .Values.server.livenessProbe.enabled }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.server.livenessProbe "enabled") "context" $) | nindent 12 }}
+            httpGet:
+              path: /-/healthy
+              port: http
+          {{- end }}
+          {{- if .Values.server.customReadinessProbe }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.server.customReadinessProbe "context" $) | nindent 12 }}
+          {{- else if .Values.server.readinessProbe.enabled }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.server.readinessProbe "enabled") "context" $) | nindent 12 }}
+            httpGet:
+              path: /-/ready
+              port: http
+          {{- end }}
+          {{- if .Values.server.customStartupProbe }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" .Values.server.customStartupProbe "context" $) | nindent 12 }}
+          {{- else if .Values.server.startupProbe.enabled }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.server.startupProbe "enabled") "context" $) | nindent 12 }}
+            tcpSocket:
+              port: http
+          {{- end }}
+          {{- end }}
+          {{- if .Values.server.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.server.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: config
+              mountPath: /opt/bitnami/prometheus/conf
+              readOnly: true
+            - name: data
+              mountPath: {{ .Values.server.persistence.mountPath }}
+          {{- if .Values.server.extraVolumeMounts }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.server.extraVolumeMounts "context" $) | nindent 12 }}
+          {{- end }}
+        {{- if .Values.server.thanos.create }}
+        - name: thanos-sidecar
+          image: {{ template "prometheus.server.thanosImage" . }}
+          imagePullPolicy: {{ .Values.server.thanos.image.pullPolicy }}
+          args:
+            - sidecar
+            - --prometheus.url={{ default "http://localhost:9090" .Values.server.thanos.prometheusUrl }}
+            - --grpc-address=0.0.0.0:10901
+            - --http-address=0.0.0.0:10902
+            - --tsdb.path=/prometheus/
+            {{- if .Values.server.thanos.objectStorageConfig }}
+            - --objstore.config=$(OBJSTORE_CONFIG)
+            {{- end }}
+            {{- if .Values.server.thanos.extraArgs }}
+            {{ toYaml .Values.server.thanos.extraArgs | indent 12  | trim }}
+            {{- end }}
+          {{- if .Values.server.thanos.objectStorageConfig }}
```

Comment:
I'd rather check tha objectStorageConfig.secretName is set and define both secretName and secretKey in values.yaml, so they get an entry in the README.md file